core/src/validator.rs

```diff
@@ -1578,7 +1578,10 @@ impl Validator {
                     .local_addr()
                     .expect("failed to get local address")
                     .port();
-                let (rtx, sender) = XdpRetransmitter::new(xdp_config, src_port)
+                let IpAddr::V4(src_ip) = node.bind_ip_addrs.active() else {
```

Comment:
Hm this is not great, because effectively it makes passing `--bind-address`
mandatory even in the non-bond case.

That's because bind-address defaults to `0.0.0.0`, which is not something we
should set as the source address in packets we send out.  

We could special case `Ipv4Addr::UNSPECIFIED` here, and we could try to see if
the given interface is a bond slave via sysfs, fetch the master, and
automatically grab the master's ip.

````
ls -l /sys/class/net/enp65s0f1/master
lrwxrwxrwx 1 root root 0 Nov 11 06:32 /sys/class/net/enp65s0f1/master -> ../../../../../virtual/net/bond0
````

sooo we can open `/sys/class/net/<iface>/master/ifindex` and if it exists we
open the device with `NetworkDevice::from_if_index` and we get the IP. 

Much success.